packages/eslint-plugin/src/rules/wrap-schedule-instead-of-ctx-schedule-rule.ts

```diff
@@ -0,0 +1,89 @@
+import * as estree from 'estree'
+import { Rule } from 'eslint'
+import { checkCallExpressionNode } from '../shared'
+
+const importsMap = {
+  wrap: 'import { wrap } from "@reatom/framework";\n',
+  schedule: 'import { schedule } from "@reatom/framework";\n',
+}
+
+const getTextToReplace = (nText: string, callbackText: string) => {
+  if (Boolean(nText)) {
+    return `schedule(ctx, ${callbackText}, ${nText})`
+  }
+  return `wrap(ctx, ${callbackText})`
+}
+
+const getMessage = (n?: estree.Expression | estree.SpreadElement) => {
+  if (Boolean(n)) {
+    return "Use 'schedule(ctx, cb, n)' instead of deprecated 'ctx.schedule(cb, n)'."
+  }
+
+  return "Use 'wrap(ctx, cb)' instead of deprecated 'ctx.schedule(cb, n)'."
+}
+
+const isImportICareAbout = (node: estree.ImportDeclaration) => {
+  return node.specifiers.some(
+    (specifier) =>
+      specifier.type === 'ImportSpecifier' &&
+      (specifier.imported.name === 'wrap' || specifier.imported.name === 'schedule') &&
+      node.source.value === '@reatom/framework',
+  )
+}
+
+export const deprecateCtxScheduleRule: Rule.RuleModule = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: "Method 'ctx.schedule' is deprecated in v4",
+    },
+    fixable: 'code',
+    hasSuggestions: true,
+    schema: [],
+  },
+  create(context) {
+    let hasImport = false
+    let lastImport: estree.ImportDeclaration | null = null
+
+    return {
+      ImportDeclaration(node: estree.ImportDeclaration) {
+        lastImport = node
+        if (isImportICareAbout(node)) {
+          hasImport = true
+        }
+      },
```

Comment:
I tried by that not works for me. 
I have no idea what's wrong and chatGpt don't help me.

Error trace:
```bash
hrow new SyntaxError(`Syntax error in selector "${rawSelector}" at position ${err.location.start.offset}: ${err.message}`);
                  ^
SyntaxError: Syntax error in selector "ImportSpecifier[source.value=/^@reatom//][imported.name=/^(wrap|schedule)$]" at position 39: Expected " ", "]", or [imsu] but "/" found.
Occurred while linting <input>
    at tryParseSelector (/*/reatom/node_modules/eslint/lib/linter/node-event-generator.js:196:19)
    at parseSelector (/*/reatom/node_modules/eslint/lib/linter/node-event-generator.js:214:28)
    at /*/reatom/node_modules/eslint/lib/linter/node-event-generator.js:265:30
    at Array.forEach (<anonymous>)
    at new NodeEventGenerator (/*/reatom/node_modules/eslint/lib/linter/node-event-generator.js:264:30)
    at runRules (/*/reatom/node_modules/eslint/lib/linter/linter.js:1077:32)
    at Linter._verifyWithoutProcessors (//reatom/node_modules/eslint/lib/linter/linter.js:1330:31)
    at Linter.verify (/*/reatom/node_modules/eslint/lib/linter/linter.js:1428:57)
    at runRuleForItem (/*/reatom/node_modules/eslint/lib/rule-tester/rule-tester.js:689:35)
    at testValidTemplate (/*/reatom/node_modules/eslint/lib/rule-tester/rule-tester.js:749:28)
    
    ````

Reply:
Unfortuanlly, no.
Is that necessary to use selector instead of if statement? 
I don't really good at eslint plugins. I don't undestand perfomance difference. 

I think my writed code is more flexible and expandable in a way to add new ImportSpecifiers just into array. Espcially if some juniors come to help with plugins.
What do you think?